### ISSUE TITLE
trackEvent's state.history grows forever, takes up memory

### ISSUE THREAD

**Issue by amedee-os** [DavidWells/analytics#358]
Hi, thanks for analyticsjs!

When using it on in a long-running server calling `trackEvent` after every request, the memory usage of the process grows forever. This shows the process's growing memory use until OOM killed. (On ~1/31 we deployed the fix below).

<img width="790" alt="Screenshot 2023-02-02 at 12 47 15 PM" src="https://user-images.githubusercontent.com/80348231/216402055-64af381e-5a3e-444b-8310-e50861c566dc.png">

 We found that the `state.history` variable of the [trackReducer](https://github.com/DavidWells/analytics/blob/master/packages/analytics-core/src/modules/track.js#L12) keeps a log of every single trackEvent sent. Adding this line after our trackEvent() call made the memory issue go away:

```javascript
  analytics.getState("track").history = [];
```

When reading the code, I can't seem to find where `state.history` is used, or if it's for external code like plugins to read.
In our use case we never read from it, so disabling it would be best, or keeping it capped at a small N if it's used for retries.

I noticed the [trackEvent history code ](https://github.com/DavidWells/analytics/blob/master/packages/analytics-core/src/modules/track.js#L28)has the comment `//Todo prevent LARGE arrays`. Could we add some kind of config to set a max history length, or to turn it off entirely? Or hard code a limit like 10k?

This might also help memory usage for browser users of analyticsjs with long running sessions.

Would you take a PR to fix this? Which approach would you prefer?

**Comment by matjam** [comment]
Nice catch. We're seeing this also.

**Comment by matjam** [comment]
Yeah, shipped that workaround but it's not working for us, I'm still seeing this. I guess I'm going to need to actually start digging a bit deeper.

![image](https://user-images.githubusercontent.com/578676/229225172-df172249-fa56-4547-a7b5-5d2ac9864e66.png)

You can see its already ticking up at the end. Pre adding the library this service was stable memory wise

![image](https://user-images.githubusercontent.com/578676/229225356-88f2d5b3-1aeb-492a-86e8-a38e8db103b5.png)

The drop on 3/29 is from us doubling the container limits.

**Comment by DavidWells** [comment]
Are your instances running serverside in a long running mode process (like express or something)?

we might be able to just disable this functionality on the serverside. It’s meant for clients that usually have short & independent sessions
